### PR TITLE
[libc] Add `link.h` and `elf.h` headers

### PR DESCRIPTION
**Pull request by izaakschroeder**
Headers for https://github.com/llvm/llvm-project/pull/97504

### CODE REVIEW

**Review by SchrodingerZhu — APPROVED** [review]
LGTM with some nits

File: libc/include/llvm-libc-types/CMakeLists.txt [hunk 1]
```diff
@@ -89,6 +89,9 @@ add_header(thrd_t HDR thrd_t.h DEPENDS .__thread_type)
 add_header(tss_t HDR tss_t.h)
 add_header(tss_dtor_t HDR tss_dtor_t.h)
 add_header(__atexithandler_t HDR __atexithandler_t.h)
+add_header(Dl_info HDR Dl_info.h)
+add_header(struct_dl_phdr_info HDR struct_dl_phdr_info.h)
+add_header(__dl_iterate_phdr_callback_t HDR __dl_iterate_phdr_callback_t.h)
```
[the remark this answers is not in the document]
Comment (SchrodingerZhu) [reply libc/include/llvm-libc-types/CMakeLists.txt:94]: also there are other dependencies missing.

[the remark this answers is not in the document]
Comment (izaakschroeder) [reply libc/include/llvm-libc-types/CMakeLists.txt]: Sorry, not all that familiar with cmake and this setup. Did my best, let me know if you think I forgot any others 😄

**Review by SchrodingerZhu — APPROVED** [review]
Thank you for the patch!

**Pull request merged**